Compression UG edits.

dwelsch-esi opened the pull request:
Style and grammar edits of the Compression User Guide.

quic-ipendse committed:
Quant Stats Visualizer - Adding Box Plot visualization for Advanced Stats

* Adding boxplots for advanced stats and modifying the backend for the same
* Cleanup JavaScript callbacks by packaging repetitive code as a function in utils
* Generalize the code in callbacks by adaptively iterating over columns present in datasources instead of manually listing them
* Dynamically adjust boxplot width according to the number of boxplots
* Update docstrings

---------

Signed-off-by: Ishan Pendse <quic_ipendse@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-hitameht committed:
Ga build docker images (#3351)

* Implement pyproject.toml to build AIMET package

There are 3 dynamic fields in metadata:
 - version
 - dependencies
 - name (not PEP compatible!)
A plugin of scikit-build-core build system generates dependencies
and package name based on `CMAKE_ARGS` environment variable.

Signed-off-by: Evgeny Mironov <quic_emironov@quicinc.com>

* Build all docker images from the single Dockerfile

Docker images contain dependencies to build and run tests.

Signed-off-by: Evgeny Mironov <quic_emironov@quicinc.com>

* Build docker images on a CI

Signed-off-by: Evgeny Mironov <quic_emironov@quicinc.com>

---------

Signed-off-by: Evgeny Mironov <quic_emironov@quicinc.com>
Co-authored-by: Evgeny Mironov <quic_emironov@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

dwelsch-esi committed:
Style and garmmar edits of the Compression User Guide.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-rgite committed:
Capture all static activation tensors for quantization (#3355)

Signed-off-by: Raj Gite <quic_rgite@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-rgite committed:
Exception rule for Gather op (#3356)

Signed-off-by: Raj Gite <quic_rgite@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Soft-deprecate rounding_mode parameter (#3358)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-klhsieh committed:
Enable seq mse with bq/lpbq

Signed-off-by: Kevin Hsieh <quic_klhsieh@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-hitameht commented on Docs/user_guide/spatial_svd.rst:
```diff
@@ -2,15 +2,22 @@
+#################
 
-Spatial SVD is a tensor decomposition technique which decomposes one large layer (in terms of mac or memory) into two smaller layers. SVD stands for Singular Value Decomposition.
+Spatial singular value decomposition (SSVD) is a technique that decomposes one large convolution (Conv) MAC or memory layer into two smaller layers.
```
Can we not abbreviate Spatial SVD to SSVD?

quic-hitameht commented on Docs/user_guide/spatial_svd.rst:
```diff
@@ -2,15 +2,22 @@
+- ℎ is the height of the kernel
+- 𝑤 is the width of the kernel 
+  
+SSVD decomposes the kernel into two kernels, one of size (𝑚,𝑘,ℎ,1) and one of size (𝑘,𝑛,1,𝑤), where 𝑘 is called the `rank`. The smaller the value of 𝑘, the larger the degree of compression.
```
Same comment as above. Please replace SSVD w/ Spatial SVD.

quic-hitameht commented on Docs/user_guide/spatial_svd.rst:
```diff
@@ -2,15 +2,22 @@
+  
+SSVD decomposes the kernel into two kernels, one of size (𝑚,𝑘,ℎ,1) and one of size (𝑘,𝑛,1,𝑤), where 𝑘 is called the `rank`. The smaller the value of 𝑘, the larger the degree of compression.
+
+The following figure illustrates how SSVD decomposes both the output channel dimension and the size of the Conv kernel itself. 
```
Same comment as above. Please replace SSVD w/ Spatial SVD.

quic-hitameht commented on Docs/user_guide/greedy_compression_ratio_selection.rst:
```diff
@@ -2,53 +2,64 @@
 ========
-The model compression methods, Spatial SVD and Channel Pruning work on per layer basis. Not all the layers in the given model are equally compressible. Compression of individual layers of a given model can have varying impact on the final accuracy of the model. Greedy Per Layer Compression Ratio Selection Algorithm is used to assess the sensitivity of applicable layers to compression and find app ...
+
+Spatial SVD (SSVD) and channel pruning (CP) work on individual layers of a model. Not all the layers are equally compressible, so compression of a given layer has a variable impact on the final model accuracy. The greedy per-layer compression ratio selection algorithm assesses the sensitivity of layers to compression and finds an appropriate compression ratio for each layer. The algorithm ensures ...
```
Same comment here. replace Spatial SVD (SSVD) w/ Spatial SVD.

dwelsch-esi committed:
Quantization User Guide edits (#3348)

* Edited Quantization User Guide.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

* Quantization Guide - more edits.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

* Corrected TOC errors introduced by Quant UG edits.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

* Review changes PR #3348 - Quantization UG edits.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

* More review changes PR #3348 - Quantization UG edits.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

---------

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Add AutoQuant with AMP (#3361)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Declare explicit namespace package (#3364)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Use formal type comparison (#3365)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-klhsieh committed:
Update RMSNorm op definition

Signed-off-by: Kevin Hsieh <quic_klhsieh@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-gsaichai committed:
Fix Matmul input order flipping in connected graph (#3367)

Signed-off-by: Sai Chaitanya Gajula <quic_gsaichai@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-ykota committed:
Explicitly disable quantizer if tensor is not float (#3368)

- Backend doesn't expect non-float tensors to have encodings. Compare op's
  output is a tensor of bools for which the encoding shouldn't have been
  present. This change fixes this issue by disabling the quantizer.

Signed-off-by: yathindra kota <quic_ykota@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Fix typos in bq docstrings

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Use untrained resnet to preclude network connection during unit test (#3370)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Add warning for the future change in the default value of quant_scheme (#3371)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-pdangi committed:
Enable batch norm fold in quantsim v2 (#3372)

* Add batch norm fold support in QuantSim v2

Signed-off-by: Priyanka Dangi <quic_pdangi@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-ykota committed:
Add manual mixed precision APIs (#3374)

Signed-off-by: yathindra kota <quic_ykota@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Migrate quantizer group unit tests (#3375)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-kyunggeu committed:
Migrate v1 code under aimet_torch.v1 subpackage (#3376)

Signed-off-by: Kyunggeun Lee <quic_kyunggeu@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Refactor onnx quantsim export

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-hitameht committed:
Fix node name within if node (#3378)

* Remove marked_module from input and output field of onnx.GraphProto

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

* Add a unit test to verify the fix

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

---------

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Clean up QcQuantizeOp encoding APIs

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-cgulecha committed:
Fixed issue in op name retrival (#3383)

Signed-off-by: Chetan Gulecha <quic_cgulecha@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Support 1.0.0 encoding format in aimet_onnx

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Resolve pylint failures

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-hitameht committed:
Add tie observers utility in aimet onnx (#3387)

* Add utility to tie quantizers in aimet_onnx

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

* Incorporate review feedback

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

* Expose temp API to apply constraints

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

---------

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-mtuttle committed:
Revert sim._export_encodings API to fix acceptance test

Signed-off-by: Michael Tuttle <quic_mtuttle@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-klhsieh committed:
Add deprecation warning for encoding version

Signed-off-by: Kevin Hsieh <quic_klhsieh@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-bharathr committed:
Unpinned torch version for aimet onnx variants (#3382)

* Delete symlinks, Unpin torch version, upgrade cuda version to 11.8

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

* Add complete path to html with --find-links

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>

* Fixed unpinned torch install in onnx dockerfiles, updated ALL dockerfiles for consistency

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>

* Pinned back numpy in onnx dev dockers, updated numpy version in reqs file

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>

* Cleaned up onnx depndency lists

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>

* Added missing files

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>

* Updated reqs_deb_onnx_gpu file with cuda 11.8 versions of packages

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>

---------

Signed-off-by: Hitarth Mehta <quic_hitameht@quicinc.com>
Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>
Co-authored-by: Hitarth Mehta <quic_hitameht@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-klhsieh committed:
Keep AIMET custom modules as a single op in connected graph

Signed-off-by: Kevin Hsieh <quic_klhsieh@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-ykota committed:
MMP: Ingest user requests and preprocess (#3394)

Signed-off-by: yathindra kota <quic_ykota@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-klhsieh committed:
Update Pytorch AMP supported kernels logic

Signed-off-by: Kevin Hsieh <quic_klhsieh@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

dwelsch-esi committed:
Replaced abbreviation "SSVD" with "spatial SVD" per review comment.

Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

quic-bharathr committed:
Fixed torch and ONNX installation instructions (added missing find-links option) (#3397)

* Added missing find-links URL topion to torch installation instructions
* Fixed onnx wheel instructions, added common pre-requisites
* Fixed onnx-cpu wheel install instructions
* Added pip version pinning

Signed-off-by: Bharath Ramaswamy <quic_bharathr@quicinc.com>
Signed-off-by: Dave Welsch <dwelsch@expertsupport.com>

dwelsch-esi committed:
Merge branch 'develop' into compress-ug-1

Signed-off-by: Dave Welsch <116022979+dwelsch-esi@users.noreply.github.com>
